src/whip-client.c

```diff
@@ -321,6 +322,27 @@ static void whip_options(void) {
 	WHIP_LOG(LOG_INFO, "\n");
 }
 
+static gboolean source_events(GstPad* pad,
+														GstObject* parent,
+														GstEvent* event) {
+	gboolean ret;
+
+	switch (GST_EVENT_TYPE(event)) {
+		case GST_EVENT_EOS:
+			ret = gst_pad_event_default (pad, parent, event);
+			// g_print("source_events:GST_EVENT_EOS\n");
+			whip_disconnect("Shutting down");
```

Comment:
Maybe you can add an ` (EOS)` after "Shutting down" so that it's clearer in the logs as well what caused a shutdown?